apps/debug/sysdbgapp_init.c

```diff
@@ -0,0 +1,228 @@
+		g_sysdbg_handle = open("/dev/sysdbg", O_RDWR);
+		if (g_sysdbg_handle < 0) {
+			printf("sysdbg: can not open /dev/sysdbg\n");
+		}
+	}
+	return g_sysdbg_handle;
+}
+
+/****************************************************************************
+ * Name:  sysdbg_dev_read
+ *
+ * Description:
+ *   This function reads sysdbg device
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   int - Negative value indicates that read failure, non negative
+ *   indicate sysdbg device is read successfully
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+static void sysdbg_dev_read(void)
+{
+	int ret = -1;
+	char *buffer = NULL;
+	if (g_sysdbg_handle < 0) {
+		/* Open the /dev/sysdbg device and read device data */
+		if (sysdbg_dev_open() < 0) {
+			return;
+		}
+	}
+	/* Passing null buffer as sysdbg device is not using this buffer to pass
+	 * data to user space. It merely prints the output using lldbg */
+	ret = read(g_sysdbg_handle, buffer, strlen(buffer));
+	if (ret < 0) {
+		printf("sysdbg: Couldn't read the device\n");
+	}
+}
+
+/****************************************************************************
+ * Name: sysdbg_dev_close
+ *
+ * Description:
+ *   This function closes sysdbg device and initialize the device handle to -1
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *  None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+static void sysdbg_dev_close(void)
+{
+	int ret = -1;
+	if (g_sysdbg_handle < 0) {
+		printf("sysdbg:  /dev/sysdbg device is not yet open\n");
+	} else {
+		ret = close(g_sysdbg_handle);
+		if (ret < 0) {
+			printf("sysdbg: failed to close /dev/sysdbg\n");
+		} else {
+			g_sysdbg_handle = -1;
+		}
+	}
+}
+
+/****************************************************************************
+ * Name:  show_usage
+ *
+ * Description:
+ *   This function prints the usage of sysdbg tash command
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+
+static void show_usage(FAR const char *progname)
+{
+	printf("USAGE: \n");
+	printf("kindly provide 2 arguments\n");
+	printf("For help, %s help\n", progname);
+}
+
+/****************************************************************************
+ * Name:  sysdbgapp_main
+ *
+ * Description:
+ *   This function handles the user inputs and writes it to sysdbg device
+ *
+ * Inputs:
+ *   argc - Number of arguments passed
+ *   argv - Array of character pointers
+ *
+ * Return Value:
+ *   int - zero indicates successful execution
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+
+int sysdbgapp_main(int argc, char **args)
+{
+	int i, nbytes;
+	char arg_str[128];
+	if (argc < 2) {
+		show_usage(args[0]);
+	} else if (strcmp("read", args[1]) == 0) {
```

Comment:
how about use strncmp to prevent security issue on code analyze tool?

Reply:
done